pkg/controller/certificaterequests/acme/acme_test.go

```diff
@@ -51,13 +50,15 @@ var (
 	fixedClock      = fakeclock.NewFakeClock(fixedClockStart)
 )
 
-func generateCSR(t *testing.T, secretKey crypto.Signer) []byte {
-	asn1Subj, _ := asn1.Marshal(pkix.Name{
-		CommonName: "test",
-	}.ToRDNSequence())
+func generateCSR(t *testing.T, secretKey crypto.Signer, extraDNSNames ...string) []byte {
+	// The CommonName of the certificate request must also be present in the DNS
+	// Names.
 	template := x509.CertificateRequest{
-		RawSubject:         asn1Subj,
+		Subject: pkix.Name{
+			CommonName: "example.com",
+		},
 		SignatureAlgorithm: x509.SHA256WithRSA,
+		DNSNames:           append(extraDNSNames, "foo.com"),
```

Comment:
This seems bizarre.. why set the CN to `example.com`, and then always append `foo.com` to the dnsNames? This seems like non-obvious behaviour. I'd expect to see `example.com` put in both the CN and dnsNames field if it is being set here, or otherwise changing `generateCSR` to accept `commonName string, dnsNames ...string` maybe?